modules/afmongodb/afmongodb-grammar.ym

```diff
@@ -88,11 +88,7 @@ afmongodb_option
     ;
 
 afmongodb_legacy_option
-    : KW_IFDEF
-        {
-#if SYSLOG_NG_ENABLE_LEGACY_MONGODB_OPTIONS
```

Comment:
I removed this, because there is no way to access this rule, `KW_IFDEF` does not have a string representation in our `parser.c` files. It cannot be used currently, dead code. We can add a string representation in a `parser.c` file, if we intend to use it.

Reply:
I don't think the purpose of KW_IFDEF is to be reachable. It is just a dummy token that is intentionally cannot be reached from config. Its purpose is to wrap around the compiler switch that ends up in the generated c file. The compiler switch switches out a block of code, even if KW_IFDEF is not runtime-reachable. It is a compile time logic.

Reply:
Wow, magic! :D I see now, thanks for the explanation. I will remove this change, and workaround this in another way.

Reply:
I fixed it with `BisonGraph:get_parents(node)` and `ConfigOptions:_remove_ifdef(graph)`.